server/handlers/decoding.go

```diff
@@ -20,12 +20,18 @@ import (
 func DecodingLimitsHandler(handler http.Handler, maxLength, gzipMaxLength int64) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		// Reject too-large requests before doing any further processing.
-		// Note(philipc): This likely does nothing in the case of "chunked"
+		// Note(philipc): This does nothing in the case of "chunked"
 		// requests, since those should report a ContentLength of -1.
 		if r.ContentLength > maxLength {
 			writer.Error(w, http.StatusBadRequest, types.NewErrorV1(types.CodeInvalidParameter, types.MsgDecodingLimitError))
 			return
 		}
+		// For chunked rqeuests (where full size is not known in advance),
+		// pass server.decoding.max_length down, using the request context.
+		if r.ContentLength < 0 || (r.ContentLength == 0 && r.Body != nil) {
+			ctx := util_decoding.AddServerDecodingMaxLen(r.Context(), maxLength)
```

Comment:
This is where the new context key is inserted. Note that we only add it when we have an "indeterminate body size" case, such as what happens during chunked requests.

Reply:
Is this only the case for chunked requests or are there some other use cases? If only true for chunked requests, can we just check the header (ie. `'Transfer-Encoding: chunked'`)?
 
Also it would be helpful to add a comment which explain the 2 conditions in the `if` statement.